CI: Give each test case a tmpdir data directory.

We recently experienced a CI failure where the failures appeared to be related the backend creating & listing spaces. @jameskerr speculated it could have been caused by not cleaning the data directory fast enough before the next starts. If that's the case, he figured we can improve the stability this by giving each test case a tmpdir data directory.

Excerpt from the full [logs.txt](https://github.com/brimsec/brim/files/5726182/logs.txt from the failed run:

```
2020-12-21T18:30:25.8047100Z test creating a subspace ... [32mok[39m [90m[3m(314ms)[23m[39m
2020-12-21T18:30:25.8048050Z 
2020-12-21T18:30:25.8065622Z failures:
2020-12-21T18:30:25.8068692Z 
2020-12-21T18:30:25.8070065Z listing the space
2020-12-21T18:30:25.8099580Z conflict with pending operation: space with name 'space1' already exists
2020-12-21T18:30:25.8100696Z     at createError (file:///D:/a/brim/brim/zealot/dist/index.es.js:606:33)
2020-12-21T18:30:25.8101608Z     at promise (file:///D:/a/brim/brim/zealot/dist/index.es.js:668:55)
2020-12-21T18:30:25.8102302Z     at async spaces_test.ts:4:3
2020-12-21T18:30:25.8102924Z     at async withZqd (test_api.ts:90:5)
2020-12-21T18:30:25.8103910Z     at async asyncOpSanitizer (deno:runtime/rt/40_testing.js:36:7)
2020-12-21T18:30:25.8106989Z     at async Object.resourceSanitizer [as fn] (deno:runtime/rt/40_testing.js:70:7)
2020-12-21T18:30:25.8108446Z     at async TestRunner.[Symbol.asyncIterator] (deno:runtime/rt/40_testing.js:245:13)
2020-12-21T18:30:25.8109580Z     at async Object.runTests (deno:runtime/rt/40_testing.js:322:22)
2020-12-21T18:30:25.8110347Z     at async $deno$test.ts:8:1
2020-12-21T18:30:25.8110752Z 
2020-12-21T18:30:25.8111189Z creating a space
2020-12-21T18:30:25.8111898Z AssertionError: Test case is leaking async ops.
2020-12-21T18:30:25.8112464Z Before:
2020-12-21T18:30:25.8112934Z   - dispatched: 243
2020-12-21T18:30:25.8113393Z   - completed: 242
2020-12-21T18:30:25.8113826Z After:
2020-12-21T18:30:25.8114275Z   - dispatched: 253
2020-12-21T18:30:25.8114731Z   - completed: 253
2020-12-21T18:30:25.8115032Z 
2020-12-21T18:30:25.8115693Z Make sure to await all promises returned from Deno APIs before
2020-12-21T18:30:25.8116381Z finishing test case.
2020-12-21T18:30:25.8117172Z     at assert (deno:runtime/rt/06_util.js:33:13)
2020-12-21T18:30:25.8118334Z     at asyncOpSanitizer (deno:runtime/rt/40_testing.js:46:7)
2020-12-21T18:30:25.8119362Z     at async Object.resourceSanitizer [as fn] (deno:runtime/rt/40_testing.js:70:7)
2020-12-21T18:30:25.8120475Z     at async TestRunner.[Symbol.asyncIterator] (deno:runtime/rt/40_testing.js:245:13)
2020-12-21T18:30:25.8121450Z     at async Object.runTests (deno:runtime/rt/40_testing.js:322:22)
2020-12-21T18:30:25.8122107Z     at async $deno$test.ts:8:1
2020-12-21T18:30:25.8122425Z 
2020-12-21T18:30:25.8122832Z failures:
2020-12-21T18:30:25.8123113Z 
2020-12-21T18:30:25.8123959Z 	listing the space
2020-12-21T18:30:25.8124436Z 	creating a space
```

--- a/src/js/components/LogDetails/Md5Panel.tsx
+++ b/src/js/components/LogDetails/Md5Panel.tsx
@@ -32,7 +32,7 @@ export const Md5Panel = ({record}: Props) => {
       .chan(3, (records) => setTx(records))
 
     return abort
-  }, [])
+  }, [logMd5])
 
   function onRightClick(field, record) {
     dispatch(contextMenu(field, record))

--- a/src/js/electron/tron/session.test.ts
+++ b/src/js/electron/tron/session.test.ts
@@ -1,17 +1,19 @@
 import fsExtra from "fs-extra"
 
 import path from "path"
+import os from "os"
 
 import disableLogger from "../../test/helpers/disableLogger"
 import formatSessionState from "./formatSessionState"
 import initTestStore from "../../test/initTestStore"
 import tron from "./"
 
-const file = "tmp/appState.json"
-disableLogger()
+const dir = path.join(os.tmpdir(), "session.test.ts")
+const file = path.join(dir, "appState.json")
 
-beforeEach(() => fsExtra.ensureDir("tmp"))
-afterEach(() => fsExtra.remove("tmp"))
+disableLogger()
+beforeEach(() => fsExtra.ensureDir(dir))
+afterEach(() => fsExtra.remove(dir))
 
 test("session loading with migrations", async () => {
   const state = initTestStore().getState()

--- a/src/js/flows/ingestFiles.test.ts
+++ b/src/js/flows/ingestFiles.test.ts
@@ -1,5 +1,4 @@
 import {createZealotMock} from "zealot"
-import fsExtra from "fs-extra"
 
 import Clusters from "../state/Clusters"
 import Current from "../state/Current"

--- a/src/js/flows/ingestFiles.test.ts
+++ b/src/js/flows/ingestFiles.test.ts
@@ -33,10 +33,6 @@ beforeEach(() => {
   store.dispatchAll([Clusters.add(conn), Current.setConnectionId(conn.id)])
 })
 
-afterEach(() => {
-  return fsExtra.remove("tmp")
-})
-
 describe("success case", () => {
   beforeEach(() => {
     zealot.stubStream("pcaps.post", [

